themes/cardinal_service/pdb_components/opportunities_list/src/app.jsx

```diff
@@ -0,0 +1,189 @@
+    {field: 'su_opp_location', label: 'Location', multiple: true},
+    {field: 'su_opp_dimension', label: 'Dimension', multiple: true},
+    {field: 'su_opp_pathway', label: 'Pathway', multiple: true},
+    {field: 'su_opp_placement_type', label: 'Placement Type', multiple: true},
+    {field: 'su_opp_service_theme', label: 'Service Theme', multiple: true}
+  ];
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      allItems: {},
+      activeItems: {},
+      filters: {},
+      showMoreFilters: false
+    };
+    this.onSelectChange = this.onSelectChange.bind(this);
+    this.onFormSubmit = this.onFormSubmit.bind(this);
+
+    if (window.location.search.length > 0) {
+      this.state.filters = queryString.parse(window.location.search, {arrayFormat: 'bracket'});
+    }
+  }
+
+  componentDidMount() {
+    const that = this;
+    fetch('/api/opportunties')
+      .then(response => response.json())
+      .then(jsonData => {
+        that.setState({
+          allItems: _.cloneDeep(jsonData),
+          activeItems: _.cloneDeep(jsonData)
+        }, this.onSelectChange)
+      });
+  }
+
+  onFormSubmit(e) {
+    e.preventDefault();
+    if (window.location.search.length === 0 && Object.keys(this.state.filters).length === 0) {
+      return;
+    }
+
+    let newLocation = window.location.pathname;
+    newLocation += '?';
+    Object.keys(this.state.filters).map(fieldName => {
+      this.state.filters[fieldName].map(tid => {
+        newLocation += `&${fieldName}[]=${tid}`;
+      })
+    });
+    newLocation = newLocation.replace('?&', '?');
+    window.location = newLocation;
+  }
+
+  onSelectChange(fieldName, selectedValues) {
+
+    const newState = {...this.state};
+
+    newState.activeItems = _.cloneDeep(newState.allItems);
+    delete newState.filters[fieldName];
+    if (selectedValues !== undefined && selectedValues.length > 0) {
+      newState.filters[fieldName] = selectedValues;
+    }
+
+    Object.keys(newState.activeItems).map(fieldName => {
+      let validEntities = [];
+
+      Object.keys(newState.filters).map(filterFieldName => {
+        if (fieldName === filterFieldName) {
+          return;
+        }
+
+        let filterGroup = [];
+        newState.filters[filterFieldName].map(selectedTid => {
+          const selectedItem = newState.allItems[filterFieldName].find(a => parseInt(a.id) == parseInt(selectedTid));
+          filterGroup = [...new Set(filterGroup.concat(selectedItem.items))];
+        })
+
+        if (validEntities.length === 0) {
+          validEntities = [...filterGroup];
+        }
+        else {
+          validEntities = validEntities.filter(x => filterGroup.includes(x));
+        }
+      })
+
+      if (validEntities.length > 0) {
+        newState.activeItems[fieldName].map(term => {
+          term.items = validEntities.filter(x => term.items.includes(x));
+        })
+      }
+    })
+
+    this.setState(newState);
+  }
```

Comment:
This block needs comments to what is happening here. This is complicated enough that it needs explanation.

Reply:
oh please. its clear cut isn't it. :laughing: